tools/rtpsrelay/lib/QosIndex.h

```diff
@@ -18,7 +18,8 @@ inline OpenDDS::DCPS::RepoId guid_to_repoid(const GUID_t& a_guid)
 inline GUID_t repoid_to_guid(const OpenDDS::DCPS::RepoId& a_guid)
 {
   GUID_t retval;
-  std::memcpy(&retval, &a_guid, sizeof(OpenDDS::DCPS::RepoId));
+  // static_cast since we are copying from a struct to a class.
+  std::memcpy(static_cast<void*>(&retval), &a_guid, sizeof(OpenDDS::DCPS::RepoId));
```

Comment:
The generated code for C++11 is not a "trivial type", though it is a trivially-copyable type.  memcpy should only require trivially-copyable.  I'd rather have that warning (or change the code to just assign/memcpy individual members) than insert that static_cast that just happens to silence the warning but has no other real effect on the program.